livre/core/pipeline/OutputPort.h

```diff
@@ -0,0 +1,137 @@
+/* Copyright (c) 2011-2016, EPFL/Blue Brain Project
+ *                     Ahmet Bilgili <ahmet.bilgili@epfl.ch>
+ *
+ * This file is part of Livre <https://github.com/BlueBrain/Livre>
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License version 3.0 as published
+ * by the Free Software Foundation.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this library; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+#ifndef _OutputPort_h_
+#define _OutputPort_h_
+
+#include <livre/core/types.h>
+
+namespace livre
+{
+
+/**
+ * The AsyncData class provides thread safe operations on data setting/retrieval/querying
+ * based on boost::shared_future ( boost::wait_for_any is not supported by std yet ).
```

Comment:
I think the commend about shared_future should be inside the implementation to justify your decision, but not here.